### DISCUSSION TITLE
A build from source question

### DISCUSSION THREAD

**Discussion by ChuckStarchaser** [endless-sky/endless-sky#7507] [General]
I apologize in advance if I'm posting this question in the wrong place;  I just can't seem to find where all the installing and compiling questions are.  I filed a feature request yesterday, and long story short I learned that my version is very old (0.9.8), so I went looking for the 9..16.1 version I was recommended, but I only found downloads for windows and mac.  I'm on Ubuntu, so I assume I'm expected to build from source.  I fount the link to the build instructions from the readme file in the source folder.  It told me to start by running scons, so I installed scons and ran it, and I get an error message about it not finding GL/glew.h.  I have at least three versions of the file in various places, I was able to locate, but there must be something more intelligent I could do than force-feeding one of them, I'm sure.

```
(base) dd@dd-s-box:~/Downloads/zips/endless_sky/endless-sky-0.9.16.1$ scons
scons: Reading SConscript files ...
scons: done reading SConscript files.
scons: Building targets ...
Building compilation database compile_commands.json
g++ -o build/release/main.o -c -std=c++11 -Wall -Wold-style-cast -fno-rtti -Werror -O3 -flto source/main.cpp
In file included from source/text/../Shader.h:19,
                 from source/text/Font.h:19,
                 from source/main.cpp:26:
source/text/../opengl.h:26:10: fatal error: GL/glew.h: No such file or directory
   26 | #include <GL/glew.h>
      |          ^~~~~~~~~~~
compilation terminated.
scons: *** [build/release/main.o] Error 1
scons: building terminated because of errors.
(base) dd@dd-s-box:~/Downloads/zips/endless_sky/endless-sky-0.9.16.1$
```

**Comment by MCOfficer** [comment]
Did you install all the required packages mentioned [here](https://github.com/endless-sky/endless-sky/blob/master/readme-developer.md#deb-based-distros), specifically libglew-dev?

**Comment by quyykk** [comment]
>  I'm on Ubuntu, so I assume I'm expected to build from source.

There's the AppImage you can download from the release page for various Linux distros. Download it, mark it as executable and then run it.  It should work out-of-the-box.

**Comment by ChuckStarchaser** [comment]
Many thanks both of you;  I did not see the required packages, I don't know how.  And I did try downloading the image, I just didn't know I had to manually make it executable.  Sorry I'm an eternal newbie in linux, in spite of being on it for more than a decade already.

**Comment by ChuckStarchaser** [comment]
And it worked out of the box, indeed.  :-)

**Reply by quyykk** [reply]
Nice, good to know that it works :)

**Comment by ChuckStarchaser** [comment]
And it's beautiful!  Day and night.  So much more art in the outfits, and the slower-scrolling star background give it a lot more depth.
I've done quite a bit of work in Blender3D, if you guys need any help with models.  I also used to do a lot of shader work, back in the days, in GLSL.  C++ too, but I have an allergy to libraries.

**Reply by quyykk** [reply]
We have a Discord you can join if you want to (https://discord.gg/ZeuASSx). New models or updating existing models is always appreciated, but be sure to discuss it there first (or on GitHub I guess). Off the top of my head I don't know which models haven't been reworked in the past already.

**Comment by ChuckStarchaser** [comment]
Frankly, I see nothing needing rework;  it is all pretty beautiful as is.  Perhaps I could help with new stuff.  I have had problems with discord in the past.  The first time I joined it I could not understand its interface.  A few years later I needed to use discord again, but some technical issue prevented me.  The old cargo and outfit expansions looked like screwdrivers, but that's been rectified.  As to what warp drive is supposed to look like nobody knows, except perhaps a startrek fan.  No, I don't see anything needing to be fixed.  Perhaps I could get started on some new subproject.  How about designing spaceports in 3D, and have the various landed screens accessible through doors in the main spaceport model, as in the old Privateer game?
